.github/workflows/test.yaml

```diff
@@ -73,6 +78,88 @@ jobs:
           yarn prettier
           yarn run tsc
 
+  test2:
+    name: Tox tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version-major: ['3']
+        python-version-minor: [
+            '9',
+            '10',
+            '11',
+            # '12',  # TODO: We have actual failing tests in 3.12, but commenting for simplicity for now.
+            #
+          ]
+        tox-shard:
+          [
+            'trace',
+            'trace_server',
+            'anthropic',
+            'cerebras',
+            'cohere',
+            'dspy',
+            'groq',
+            'langchain',
+            'litellm',
+            'llamaindex',
+            'mistral0',
+            'mistral1',
+            'openai',
+          ]
+      fail-fast: false
+    services:
+      wandbservice:
+        image: us-central1-docker.pkg.dev/wandb-production/images/local-testcontainer:master
```

Comment:
why this base?

Reply:
That's the container we use to test things that need gorilla.

Do you think there's a better container to use?